### ISSUE TITLE
The sinon-wrapper isn't waiting for promises

### ISSUE THREAD

**Issue by MiguelMadero** [elwayman02/ember-sinon-qunit#55]
I can probably fix this and send a PR, but I wanted to log it somewhere before I forget. 

![image](https://cloud.githubusercontent.com/assets/47388/15726546/4b951d14-2806-11e6-88f6-052c43cbb347.png)

```js
                try {
                    result = callback.apply(this, args);
                } catch (e) {
                    exception = e;
                }

                if (!doneIsWrapped) {
                    if (typeof exception !== "undefined") {
                        sandbox.restore();
                        throw exception;
                    } else {
                        sandbox.verifyAndRestore();
                    }
                }
```

I'm on a component integration test, I return a promise, I would expect things to wait, however, sinon restores all my stubs before my tests completes.

**Comment by elwayman02** [comment]
👍  this seems reasonable, we definitely don't want to break async behavior!

**Comment by hvgotcodes** [comment]
Is there a resolution for this?  It's blocking me too.

**Comment by jmar910** [comment]
Blocked on this as well, reverting to just using ember-sinon

**Comment by elwayman02** [comment]
@hvgotcodes @jmar910 @MiguelMadero sorry for the long wait! I've been on vacation the last 3 weeks and haven't had the opportunity to look into it. I'll try to make some time this week!

**Comment by elwayman02** [comment]
@hvgotcodes @jmar910 @MiguelMadero it would really help if one of you could submit a PR that adds a failing test for this use-case, so I can more easily understand the scenario and make sure we don't regress in the future after a fix is found!

**Comment by elwayman02** [comment]
Fixed by #76

**Comment by elwayman02** [comment]
If anyone is still having issues with this, it might be related to an actual sinon bug:

https://github.com/sinonjs/sinon/issues/1119
